src/components/ChangePlan/ChangePlan.js

```diff
@@ -1,58 +1,145 @@
+        bigData: (chunks) => optionItem(chunks, bigDataBullet()),
+        newBigData: (chunks) =>
+          optionItem(chunks, bigDataBullet(_('change_plan.big_data_tooltip')), newLabel()),
+      }}
+    >
+      {(txt) => <ul className="dp-list-detail">{txt}</ul>}
+    </FormattedMessage>
+  );
+};
+
+const optionItem = (chunks, bullet, label) => {
+  return (
+    <li>
+      {bullet}
+      <span>
+        {chunks} {label}
+      </span>
+    </li>
+  );
+};
 
-const ChangePlan = ({ location }) => {
+const basicBullet = () => {
+  return <span className="dp-icodot">.</span>;
+};
+
+const dopplerLabel = (text) => {
+  return <span className="dp-new">{text}</span>;
+};
+
+const bigDataBullet = (tooltipText) => {
+  return (
+    <div className="dp-tooltip-container">
+      <span className="dp-icobd">BD</span>
+      <div className="dp-tooltip-block">
+        <span className="tooltiptext">{tooltipText}</span>
+      </div>
+    </div>
+  );
+};
+
+const ChangePlan = ({ location, dependencies: { planService, appSessionRef } }) => {
   const promoCode = extractParameter(location, queryString.parse, 'promo-code') || '';
   const advancedPay = extractParameter(location, queryString.parse, 'advanced-pay') || 0;
-  // TODO: remove planId, it just for testing right now
-  const planId = parseInt(extractParameter(location, queryString.parse, 'planId')) || 0;
   const intl = useIntl();
   const _ = (id, values) => intl.formatMessage({ id: id }, values);
 
-  const [isFeaturesVisible, setIsFeaturesVisible] = useState(false);
-  // Allow preselect by email or contacts
-  const selectedType = extractParameter(location, queryString.parse, 'selected-type') || '';
-  const plans = getPlans();
+  const [state, setState] = useState({
+    loading: true,
+    isFeaturesVisible: false,
+  });
+  useEffect(() => {
+    const mapCurrentPlan = (plan, planList) => {
+      const exclusivePlan = { type: 'exclusive' };
+      switch (plan.planType) {
+        case 'demo':
+        case 'free':
+          return {
+            type: 'free',
+            subscriberLimit: 500,
+            featureSet: 'free',
+          };
+        case 'subscribers':
+        case 'monthly-deliveries':
+          // for subscribers and monthly plan will be exclusive until id plan is deployed in doppler
+          const monthlyPlan = planService.getPlanById(plan.idPlan, planList);
+          return monthlyPlan ? monthlyPlan : exclusivePlan; //if not found it is exclusive plan
+        case 'prepaid':
+          return planService.getCheapestPrepaidPlan(plan.maxSubscribers, 'standard', planList);
+        case 'agencies':
+          return {
+            type: 'agency',
+            featureSet: 'agency',
+          };
+        default: {
+        }
+      }
+    };
+    const fetchData = async () => {
+      setState({ loading: true });
+      const planList = await planService.getPlanList();
+      const currentPlan = mapCurrentPlan(appSessionRef.current.userData.user.plan, planList);
```

Comment:
Why don't get the current plan from the service instead mapping it here?

Reply:
Inside `mapCurrentPlan` function there is a _getting from the service_  by id. All the information regarding actual plan is in the header and user data.

Reply:
Ok, surely I'm missing something, but, what about to send this `appSessionRef.current.userData.user.plan` to the service and manage the mapping there? Maybe I have a wrong concept and it's not the best way to manage this in a service but I think that would be clearer.

Reply:
Maybe you are right, I will give it a try.

Reply:
Adding `mapCurrentPlan` into the service would imply mapping the session type, or taking this as an any. This is not so clean. And this session plan data has nothing to do with the plan structure we are handling, this is why I´m doing the mapping in the first place, here and outside the plan service.

Reply:
Ok. And what about move this function outside useEffect?

Reply:
If I move it outside I need to add it in the dependency array. It was a compiler suggestion to put it inside of `useEffect` hook

Reply:
Oh 😞 it's true. I'm not convinced about this but it's seems to be that we have no choice.